### ISSUE TITLE
Asking for help

### ISSUE THREAD

**Issue by SmoothieAq** [hunyadi/pyopenapi#1]
Hi, sorry, not a bug, but an somewhat unrelated question.

I am new to Python, and are trying to figure out which libraries and tools I should use to make an OpenApi compliant application. I might just be bad at browsing but I mostly find all kinds of bloated frameworks.

pyopenapi and  strong-typing seems to have a style that is what I am looking for. But seems to not be used by many? Are there alternatives I also should be looking at?

But more importantly, how do I use pyopenapi and strong-typing with something that makes it a complete application with some builtin web server? Something that also uses your @webmethod annotation to wire up the web server? But also something that allows me to serve also other media types?

Thank you very much in advance!
--Jesper

**Comment by hunyadi** [comment]
**tl;dr**: If you are new to Python, this library is likely not for you.

The primary goal for *pyopenapi* has not been building a complete application but rather creating an OpenAPI specification with a Python syntax. Internally, we have several applications, implemented in multiple programming languages (including C#, Java, Python, Scala), yet we wanted to have a consistent way to declare web operations, which is how *pyopenapi* was born. Our goal was to make active use of Python type annotations, which we found not to be fully utilized in popular web application development frameworks in Python. Specifically, we found that those frameworks wanted to solve many things, and forced us into particular (but not at all elegant) syntax and old OpenAPI versions, whereas our goal was to generate and expose an OpenAPI specification, adhering to the latest OpenAPI standard. The library *pyopenapi* lets us use code refactoring features in Visual Studio Code while we are collaborating, and generate a well-formed OpenAPI specification as an artifact in a continuous integration toolchain.

**Comment by SmoothieAq** [comment]
Well, that is how I work professional in the Java world, so good for you :-)
This is however a hoppy project, where I look for someting simpler.
Thank you very much for your clear reply!
